src/core/server/http/router/response_adapter.ts

```diff
@@ -112,8 +113,14 @@ export class HapiResponseAdapter {
     return response;
   }
 
-  private toError(kibanaResponse: KibanaResponse<ResponseError>) {
+  private toError(kibanaResponse: KibanaResponse<ResponseError | Buffer | stream.Readable>) {
```

Comment:
I think it may be more appropriate to add `Buffer | stream.Readable` to the `ResponseError` union type. @restrry WDYT?

Reply:
When it's required? As I can see proxyRequest rejects a promise with error message not a steam or a buffer https://github.com/elastic/kibana/blob/0e8fae18bd5468aa9bb6e884506ddbe1593e3e4a/src/plugins/console/server/lib/request.ts#L92

Reply:
Hey @restrry ! Thanks for weighing in!

This is the flow:

User issues request in browser -> console streams to ES -> ES responds (e.g., status code 400!) -> console streams back to browser -> user sees error.

With the current implementation console cannot stream the error response ES gave us back to browser because the current implementation wants to send back JS object encoded as JSON when we haven't read the body into memory for that.

The error handler you linked to is for lower-level errors on `http` request - like if the socket gets hung up. Not for a 4xx or 5xx status code. You can see here how the response is handled: https://github.com/elastic/kibana/blob/541ecf5d9ccd828dbef9e809fb31c6c5e268f133/src/plugins/console/server/routes/api/console/proxy/create_handler.ts#L166

Reply:
Ok, I think we can merge the current changes. Could you add headers configuration as done in https://github.com/elastic/kibana/pull/55690/files/46c5cab221d6b63a8b442f3ba8a3a7bdc4c03f4d#diff-92d94ad4d163809567c4258f95ff37e2R96 ?
I will create an issue to add tests for this API

Reply:
> I think it may be more appropriate to add Buffer | stream.Readable to the ResponseError union type. @restrry WDYT?

agree, can be done in https://github.com/elastic/kibana/issues/56305